app/src/main/java/ch/epfl/sweng/SDP/Activity.java

```diff
@@ -167,11 +167,11 @@ private void styleView(TextView view, String text, int color, int size, Typeface
     protected void cloneAccountFromFirebase(@NonNull DataSnapshot snapshot) {
         HashMap<String, HashMap<String, Object>> userEntry =
                 (HashMap<String, HashMap<String, Object>>) snapshot.getValue();
-        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
```

Comment:
Explanation: This line prevented us from testing this method properly, because it would fail getting an instance when called from a test. Since snapshot already contains the userId (the userId is the key of this snapshot), I now retrieve the ID from the snapshot.